sys/net/application_layer/nanocoap/sock.c

```diff
@@ -224,14 +225,15 @@ ssize_t nanocoap_sock_get(nanocoap_sock_t *sock, const char *path, void *buf, si
     ssize_t res;
     coap_pkt_t pkt;
     uint8_t *pktpos = buf;
+    uint16_t id = (uint16_t)random_uint32();
```

Comment:
This is actually not ideal. There are strict requirements on how often a message ID can be reused. Hence, and has to create message IDs in a way they are used round robin. Even if the 32 bit PRNG spillls out a full sequence prior to issuing the same number again, this would likely not be true when only looking at the lower 16 bit of the number.

The standard suggests to use an counter to optain the ID - ideally a different one for each endpoint to communicate with, but one global counter is fine, too. Ideally the counter is initialised with a random number at boot, so that chances of reusing the same message ID after a reboot/crash is low.

It would certainly be an improvement compared to just using 0, though.